packages/debugger/src/panels/kernelSources/filter.tsx

```diff
@@ -12,24 +12,24 @@ import { IDebugger } from '../../tokens';
  */
 export interface IFilterBoxProps {
   model: IDebugger.Model.IKernelSources;
-  onFilterChange?: (e: any) => void;
 }
 
 const FilterBox = (props: IFilterBoxProps) => {
-  /**
-   * Handler for search input changes.
-  const handleChange = (e: React.FormEvent<HTMLElement>) => {
-    const target = e.target as HTMLInputElement;
-    setFilter(target.value);
-    props.model.filter = target.value;
+  const [filter, setFilter] = useState('');
```

Comment:
You could keep the previous implementation and change the `UseSignal` L42 to listen for the new signal `filterChanged`. That will avoid duplicating filter state in the React component and in the model.

Reply:
I remember I have done that to begin and came into issues where the filter content was not rended. I have suspected some weird interaction between the debouncer and the react.js refresh mecanism. I have spent some time to understand and try to fix that. I am going to try again with fresh eyes and will report. I have not found similar cases in the jupyterlab code base so it may be the first time this case is tried out.

Reply:
oh rereading you comment... yes I had to create for a new signal otherwise the react component was not refreshing. I am going to try to use it to make the implementation cleaner.

Reply:
React component state is cleaner using that new signal. Done, Thx.